source/tutorials/kubernetes.rst

```diff
@@ -3,354 +3,657 @@
+   .. code-block:: bash
 
-.. code-block:: console
+      sudo kubeadm init \
+      --cri-socket=/run/containerd/containerd.sock
+      --pod-network-cidr 10.244.0.0/16
 
-   --pod-network-cidr 10.244.0.0/16
 
-Furthermore, if you are using CRI-O and `flannel` and you want to use Kata Containers, edit the :file:`/etc/crio/crio.conf` file to add:
+#. (Optional) By default, containerd will use the default runc as the default
+   runtime. `Kata Containers`_ runtime can be set as the runtime on a per-pod
+   basis. See the Kata Containers documentation on `creating an untrusted pod
+   <https://github.com/kata-containers/documentation/blob/master/how-to/how-to-use-k8s-with-cri-containerd-and-kata.md#create-an-untrusted-pod-using-kata-containers>`_
+   for more information.
 
-.. code-block:: console
+#. Once the cluster initialization is complete, continue reading about how to
+   `Use your cluster`_.
 
-    [crio.runtime]
-    manage_network_ns_lifecycle = true
 
-Use your cluster
-****************
+CRI+O
+-----
 
-Once your master control plane is successfully initialized, instructions on how to use your cluster and its *IP*, *token*, and *hash* values are displayed. It is important that you record the cluster values because you need them to join worker nodes to the cluster. Some values have a valid period. The values are presented in a format similar to:
+For information on CRI+O as a Kubernetes container runtime, see `What is
+CRI+O?`_. To use CRI+O as the Kubernetes container runtime:
 
-.. code-block:: bash
+#. Start the CRI-O service and enable it to run at boot automatically:
 
-   kubeadm join <master-ip>:<master-port> --token <token> --discovery-token-ca-cert-hash <hash>
+   .. code-block:: bash
 
+      sudo systemctl enable --now crio.service
 
-**Congratulations!**
+   When the crio service starts for the first time, it will create a
+   configuration file for crio at :file:`/etc/crio/crio.conf`.
 
-You've successfully installed and set up Kubernetes in |CL| using CRI-O and kata-runtime. You are now ready to follow on-screen instructions to deploy a pod network to the cluster and join worker nodes with the displayed token and IP information.
+#. Run the kubeadm command to initialize the master node with the
+   :command:`--cri-socket` parameter:
 
-Related topics
-**************
+   .. important:: 
 
-Read the Kubernetes documentation to learn more about:
+      You may need to add additional parameters to the command below,
+      depending the pod network addon in use. 
+      
+      In this example, the :command:`--pod-network-cidr 10.244.0.0/16`
+      parameter is fto use *flannel* as the pod networking. See `Choose a pod
+      network add-on`_ for more information.
 
-*  Deploying Kubernetes with a `cloud-native-setup`_
+   .. code-block:: bash
 
-*  :ref:`Kubernetes best practices <kubernetes-bp>`
+      sudo kubeadm init \
```

Comment:
Did this work for you? This is where I lost a few days trying to figure out my problem.  I had to to this for crio to work and probably also would have to do for containerd.  I am not sure about docker though.  Also, you don't have to specify the cri-socket if only one is running.  It is probably good to do anyways.  

```You need to set the kubelet to use system as the cgroup. This can be set in kubeadm.yaml as well which is more convenient then reloading systemd daemon
sudo mkdir -p /etc/systemd/system/kubelet.service.d/
cat << EOF | sudo tee  /etc/systemd/system/kubelet.service.d/10-cgroup-driver.conf
[Service]
Environment="KUBELET_EXTRA_ARGS=--cgroup-driver=systemd"
EOF
sudo systemctl daemon-reload
```

Reply:
I had the same issues as you. I'm not going to address it in this PR. See from OP
> There are still known complexities and issues with the process over all. This PR is not meant to address all issues or be the final set of instructions.

Reply:
Actually, I added this now. I remember testing it.